demisto_sdk/commands/common/docker/tests/dockerhub_client_test.py

```diff
@@ -0,0 +1,246 @@
+                "3.10.11.61265",
+                "3.8.3.9324",
+                "3.9.8.24399",
+            ],
+            "3.10.13.81631",
+        ),
+        (["1.0.0.81877", "1.0.0.78900", "1.0.0.72295"], "1.0.0.81877"),
+        (["1.0.0.81877", "1.0.0.78900", "2.0.0.72295"], "2.0.0.72295"),
+        (
+            [
+                "invalid_value",
+                "7.4.0.80528",
+                "1.5.0.80528",
+                "2.5.0.80528",
+                "2.5.0.80529",
+            ],
+            "7.4.0.80528",
+        ),
+    ],
+)
+def test_get_latest_docker_image_tag(
+    requests_mock,
+    dockerhub_client: DockerHubClient,
+    tags: List[str],
+    expected_highest_tag: str,
+):
+    """
+    Given:
+        - lists of tags
+
+    When:
+        - running get_latest_docker_image_tag method
+
+    Then:
+        - ensure that the latest tag is returned always
+    """
+    requests_mock.get(
+        "https://auth.docker.io/token",
+        json={"token": "1234", "issued_at": "1234", "expires_in": 300},
+    )
+    docker_image = "demisto/python3"
+    requests_mock.get(
+        f"{dockerhub_client.DEFAULT_REGISTRY}/{docker_image}/tags/list",
+        json={"tags": tags},
+    )
+    assert dockerhub_client.get_latest_docker_image_tag(docker_image) == Version(
+        expected_highest_tag
+    )
+
+
+@pytest.mark.parametrize(
+    "responses, count",
+    [
+        (
+            [
+                {"count": 100, "results": list(range(30)), "next": "next_link"},
+                {"count": 100, "results": list(range(30)), "next": "next_link"},
+                {"count": 100, "results": list(range(30)), "next": "next_link"},
+                {"count": 100, "results": list(range(10)), "next": None},
+            ],
+            100,
+        ),
+        (
+            [
+                {"count": 600, "results": list(range(100)), "next": "next_link"},
+                {"count": 600, "results": list(range(100)), "next": "next_link"},
+                {"count": 600, "results": list(range(100)), "next": "next_link"},
+                {"count": 600, "results": list(range(100)), "next": "next_link"},
+                {"count": 600, "results": list(range(100)), "next": "next_link"},
+                {"count": 600, "results": list(range(100)), "next": None},
+            ],
+            600,
+        ),
+        (
+            [
+                {"count": 158, "results": list(range(100)), "next": "next_link"},
+                {"count": 158, "results": list(range(58)), "next": None},
+            ],
+            158,
+        ),
+    ],
+)
+def test_do_docker_hub_get_request_with_pagination(
+    mocker,
+    dockerhub_client: DockerHubClient,
+    responses: List[Dict[str, Any]],
+    count: int,
+):
+    """
+    Given:
+        - pagination responses
+
+    When:
+        - running do_docker_hub_get_request method
+
+    Then:
+        - ensure that we retrieve all the objects eventually after pagination
```

Comment:
same here.

Reply:
also no special use-cases, mainly to validate that pagination works as expected in a different number of objects.